File: sast-python3/case/completeness/single_app_tracing/datatype/array/array_003_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->数据类型和结构->数组
# scene introduction = append操作
# level = 2+
# bind_url = completeness/single_app_tracing/datatype/array/array_003_T
# evaluation information end

import os
import array

def array_003_T(taint_src):
    char_array = array.array('u', taint_src)  # 每个字符作为独立元素

    s = array.array('u',"b")  
    s.append(char_array[0]) 
    taint_sink(s)


def taint_sink(o):
    os.system(''.join(o))


if __name__ == "__main__":
    taint_src = "taint_src_value"
    array_003_T(taint_src)

File: sast-go/cases/completeness/single_app_tracing/datatype/array/array_003_T/array_003_T.go

// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->数据类型和结构->数组
// scene introduction = 二维
// level = 2
// bind_url = completeness/single_app_tracing/datatype/array/array_003_T/array_003_T
// evaluation information end

package array_003_T

func array_003_T(__taint_src string) {
	var str = [3][1]string{[1]string{__taint_src}, [1]string{"b"}, [1]string{"c"}}
	__taint_sink(str)
}

func __taint_sink(o interface{}) {
}
